crates/bevy_asset/src/asset_server.rs

```diff
@@ -52,6 +66,7 @@ pub struct AssetServerInternal {
     pub(crate) asset_ref_counter: AssetRefCounter,
     pub(crate) asset_sources: Arc<RwLock<HashMap<SourcePathId, SourceInfo>>>,
     pub(crate) asset_lifecycles: Arc<RwLock<HashMap<Uuid, Box<dyn AssetLifecycle>>>>,
+    errors: Arc<Mutex<HashMap<HandleId, AssetServerError>>>,
```

Comment:
This will grow unbounded if nobody consumes the errors. Ideally errors are solved prior to publishing games, so maybe its ok, but we should discuss alternatives. Ex: would could use bevy's double-buffered events and assume consumers that care will poll once per frame.

Reply:
Part of my reasoning was that this is never going to get any more entries than the number of asset loads the user attempts. Because of the use case, this does not have anywhere near the same level of risk to grow unbounded as bevy events have. With events, the risk is very much real, because you can have "high volume" events emitted by systems every frame. Asset loads are very "low volume", and you can have at most 1 error per asset load attempt.

Also, considering that the user is attempting to load assets, the expectation / "happy case" would be that those assets actually get loaded ... and the actual assets would certainly take up a lot more memory than a bunch of errors. So, in practice, the amount of memory used up by these errors should always be much smaller than the memory use the user would actually expect if there were no errors.

Reply:
I think *not* using events / *not* having it auto-clear is *better*. If we used bevy events for this, it becomes unreliable and there is a risk of missing the errors. And if users are in a position to want to handle asset load errors, they'd probably expect them to work reliably / handle them all, without footguns.

Reply:
Calling `asset_server.load(path)` will return the cached handle if it has already been loaded, so it make sense to use that in a running system instead of keeping your own cache... In case of a load error this way they could pile up

Reply:
Wouldn't calling `.load(path)` repeatedly produce the same `HandleId` and hence the new error would overwrite the old one in the `HashMap`? If that's the case, then additional errors wouldn't pile up.

Reply:
my bad, you're right, the `HandleId` is made from the path so it should stay the same 👍